client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -449,9 +449,9 @@ private SendResult sendDefaultImpl(//
             String[] brokersSent = new String[timesTotal];
             for (; times < timesTotal; times++) {
                 String lastBrokerName = null == mq ? null : mq.getBrokerName();
-                MessageQueue tmpmq = this.selectOneMessageQueue(topicPublishInfo, lastBrokerName);
-                if (tmpmq != null) {
-                    mq = tmpmq;
+                MessageQueue mqAttempt = this.selectOneMessageQueue(topicPublishInfo, lastBrokerName);
```

Comment:
IMO , any variable named like XXXtmp  is bad. Maybe we can rename it to another one but tmpMq is far from good enough